### ISSUE TITLE
Tracker: arrow-layers module

### ISSUE THREAD

**Issue by ibgreen** [visgl/deck.gl-community#79]
## v9.3+

- [ ] #145
- [ ] https://github.com/visgl/loaders.gl/issues/1443

## v9.1

- [x] Fork @ibgreen 
- [ ] Docs
  - [x] Integrate into website
  - [ ] Clean up / extend
- [ ] Fix examples
- [ ]  Modernize examples
- [ ] Integrate examples into website

**Comment by charlieforward9** [comment]
To move this tracker along during 9.3...

What is the difference between this module and https://github.com/geoarrow/deck.gl-geoarrow, in writing.

@ibgreen mentioned different formats of passing the data in, but since this hasn't been very maintained, I'm curious if we should sunset it and include the geoarrow official layer like we did all the other external community layers.

Itd be nice to clear this off the backlog soon 

@kylebarron

**Comment by xinaesthete** [comment]
> To move this tracker along during 9.3...
> 
> What is the difference between this module and https://github.com/geoarrow/deck.gl-geoarrow, in writing.
> 
> [@ibgreen](https://github.com/ibgreen) mentioned different formats of passing the data in, but since this hasn't been very maintained

FWIW, I did get some very quick feedback from the author yesterday when I started trying to use this, and they immediately made a pass on some of the needed maintenance.

I would be very interested in a pointer to a good summary on this... actually, now may be a good time for me to jump onto Slack. I'm in the process of trying to implement proper support for parquet points and shapes in https://github.com/Taylor-CCB-Group/SpatialData.js

**Comment by ibgreen** [comment]
- On a high level, the long-term goal has always been to have the deck.gl ecosystem integrate Apache Arrow as a first class citizen, rather than offer a duplicated catalog of arrow layers.
- This means that deck layers/loaders.gl loaders etc will be Arrow native, luma will offer Arrow compatible GPGPU support etc.
- The arrow-layers module here in community has only been intended as a staging area to get some work done independently but this will eventually be integrated into deck etc.
- This arrow-ification effort got side tracked for a long time, but is now in full swing. 
- I am mid-stream in the first step, which is re-building loaders.gl 5.0 to be arrow-first.
- @Pessimistress is working on GPGPU support in luma.gl
- etc

We would love to have speaking partners to help guide the new architecture into something that feels right for a broad group of users. 

There is a channel #openvis-wg-binary-data on the OpenJS slack that would be a good place to get a discussion going.

**Comment by charlieforward9** [comment]
Thank you for the status update! 

And as for the https://github.com/geoarrow/deck.gl-geoarrow, I'm understanding that the processes within these layers are what you are targeting as first class? Making this repo an "archive"? Or would implementations significantly differ ?

**Comment by kylebarron** [comment]
[`@geoarrow/deck.gl-geoarrow`](https://github.com/geoarrow/deck.gl-geoarrow) (in the process of renaming from `@geoarrow/deck.gl-layers`) is a module that duplicates most existing deck.gl layers to allow passing [GeoArrow tables](https://geoarrow.org/). This is much faster than providing JSON input and is a core part of what makes [Lonboard](https://github.com/developmentseed/lonboard) so fast to render large data from Python. See ["how it works"](https://developmentseed.org/lonboard/latest/how-it-works/).

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/3074bfb1-361e-4c10-8b6e-4724a204aebc" />

> * On a high level, the long-term goal has always been to have the deck.gl ecosystem integrate Apache Arrow as a first class citizen, rather than offer a duplicated catalog of arrow layers.

I'm 100% supportive of Arrow becoming a first-class citizen and I'd be happy to one-day deprecate `@geoarrow/deck.gl-geoarrow`. 

But `deck.gl-geoarrow` is an essential stopgap that enables _today_ fast GeoArrow rendering via deck.gl's binary data interface.

I'm currently revamping `deck.gl-geoarrow` with higher code quality, nicer documentation, etc. It'll be really useful for me to present `deck.gl-geoarrow` and `deck.gl-raster` as my (opinionated) workflow to support rendering large vector and raster data in deck.gl. 


> * This means that deck layers/loaders.gl loaders etc will be Arrow native, luma will offer Arrow compatible GPGPU support etc.


I don't really have the time to spend on integrating `deck.gl-geoarrow` upstream, but feel free to ping me on Arrow-related PRs into deck/luma and I'll try to review if I have time.

**Comment by charlieforward9** [comment]
Adding an AI angle to this tracker, separated from #534 which we are closing as out of scope:

Once arrow-first lands across deck / loaders / luma per @ibgreen's roadmap above, the Arrow-native layer and loader APIs become the canonical surface that AI agents will target. Two follow-ups worth keeping on this tracker as that work progresses:

1. **LLM-first docs for Arrow-native layers and loaders** — every Arrow-native layer / loader should ship with tool-call-friendly examples in its docs (minimal input schema, columnar-data shape, explicit error modes). Agents construct layers from docs today; we can make that path straight by design rather than retrofitting.

2. **Arrow as the wire format for AI tool calls** — letting agents reference Arrow tables by ID (via DuckDB-WASM or similar in-browser analytics stacks) avoids the Arrow → JSON → Arrow round-trip that current LLM tool-call schemas force. Becomes load-bearing as ecosystems like [sqlrooms](https://sqlrooms.org), [lonboard](https://github.com/developmentseed/lonboard), and [deck.gl-geoarrow](https://github.com/geoarrow/deck.gl-geoarrow) converge.

Happy to open the Arrow-AI angle once the ball starts rolling.

**Comment by charlieforward9** [comment]
Ib is proposing https://github.com/visgl/deck.gl/pull/10367. The eventual goal for the `editable-layers` module to follow suite.

Inline with the direction of deck/loader, I believe a significant improvement would be [adopting arriw](https://github.com/visgl/loaders.gl/issues/3316#:~:text=Binary%20and%20columnar,across%20loaders.gl). for data & onEdit. While maintaining the geojson handlers for existing users and simpler, lighter use cases.

I havent worked with arrow yet, but my top questions are around the math within each mode that works with geojson by default - so would this need a arrow->geojson adapter? 

My goal is to boost the attractiveness of merging this into deck v10 ecosystem instead of keeping them separate. Emphasizing the maturity of the module and promoting more experimental contributions in this community repo.

This work would justify a proactive alias to EditableLayer in v9.4 with a migration warning to deck.gl/editable-layers in v10.

Thoughts? @eKerney @ibgreen @kylebarron @chrisgervang

**Comment by kylebarron** [comment]
FWIW I'm not sure Arrow makes sense as an internal data structure for an editable layer. Arrow chunks are immutable, so you'd have to re-allocate full Arrow chunks whenever an edit comes in, or manage an [Iceberg](https://iceberg.apache.org/)-like edit log and compaction.

**Comment by charlieforward9** [comment]
Closing in favor of https://github.com/visgl/deck.gl/issues/10270
